my-movie-list/src/components/SearchForm/index.tsx

```diff
@@ -1,25 +1,38 @@
 import React from "react";
-import { searchMovies } from "../../services/omdb";
+import { searchMovies, searchMoviesById } from "../../services/omdb";
+import "./index.css";
 
 type SearchFormProps = {
   search: AppState["search"];
   setSearch: (search: string) => void;
-  setMovies: (movies: Movie[]) => void;
+  setSearching: (searching: boolean) => void;
+  setSearchErro: (erro: boolean) => void;
+  setMoviesResultSearch: (movies: Movie[]) => void;
 };
 
-export const SearchForm = ({
-  search,
-  setMovies,
-  setSearch,
-}: SearchFormProps) => {
-  const handleSubmit = (e: React.FormEvent<HTMLFormElement>) => {
+export const SearchForm = ({ search, setMoviesResultSearch, setSearch, setSearching, setSearchErro }: SearchFormProps) => {
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
     e.preventDefault();
 
     if (!search) return;
+    setMoviesResultSearch([]);
+    setSearching(true);
+    let moviesFromAPI = await searchMovies(search);
+
+    if (moviesFromAPI.Error) {
+      setSearchErro(true);
+      setSearching(false);
+      return;
+    }
+    setSearchErro(false);
+    let movies: Movie[] = [];
 
-    searchMovies(search)
-      .then((moviesFromAPI) => setMovies(moviesFromAPI.Search))
-      .catch(console.error);
+    Promise.all(
+      moviesFromAPI.Search.map(async (movie: Movie) => movies.push(await searchMoviesById(movie.imdbID)))).then(_ => {
+        setMoviesResultSearch(movies);
+        setSearching(false);
+      });
```

Comment:
O que acha de fazer assim?
```suggestion
    Promise.all(
      moviesFromAPI.Search.map(async (movie: Movie) =>
        searchMoviesById(movie.imdbID)
      )
    ).then((movies) => {
      setMoviesResultSearch(movies as Movie[]);
      setSearching(false);
    });
```

Aí não precisa do `let movies: Movie[] = []`